### PR TITLE
Migration to logstash-input-netflow

### PR DESCRIPTION
**Pull request by jorritfolmer**
From logstash-codec-netflow.

### CODE REVIEW

**Comment by untergeek** [comment]
Wow!  What a ton of work!  70 commits!

Can you squash them, please?

**Comment by untergeek** [comment]
Even though this is based on the netflow codec, it's a new plugin.  We can probably start the versioning at 1.0.0

File: NOTICE.TXT [hunk 1]
```diff
@@ -0,0 +1,5 @@
+Elasticsearch
+Copyright 2012-2015 Elasticsearch
```
Comment (untergeek) [inline NOTICE.TXT]: Let's change the copyright to 2016, please

**Comment by jorritfolmer** [comment]
Wouldn't it be a good idea to keep the history intact instead of squashing?
The 70 commits are the entire history of the netflow plugin, the port to logstash-input-netflow is only 4 commits.

**Comment by jorritfolmer** [comment]
I'll fix the cp and version

**Comment by untergeek** [comment]
It's debatable.  We'll still have the history in the netflow codec, as the repository isn't likely to go away.

**Comment by jorritfolmer** [comment]
Ok done

**Comment by alaamallah** [comment]
Hey,

I am testing the logstash-input-netflow, but for some reason I am getting no matching template for flow id 516, however when I run the logstash using the logstash-codec-netflow, with same config file it works fine.
Here is how my config file looks like:

 cat /etc/logstash/conf.d/logstash-input-netflow.conf
input {
  netflow {
    host => "localhost"
    port => 4739
    versions => [10]
    target => ipfix
            ipfix_definitions => "/usr/share/logstash/vendor/bundle/jruby/1.9/gems/logstash-input-netflow/lib/logstash/inputs/netflow/ipfix.yaml"
    }
  }

output {
   elasticsearch {
      hosts => ["localhost:9200"]
       }
   }

Any thoughts? Wonder if it's been tested before.
Thanks a lot.
Alaa

**Comment by acchen97** [comment]
@suyograo @untergeek should we proceed with moving the codec into this input plugin?

**Comment by debugloop** [comment]
@jorritfolmer worked out that https://github.com/logstash-plugins/logstash-codec-netflow/issues/71 is due to a template being sent by two routers with the same template and source id. As logstash-input-netflow would have access to the host field this issue could be solved easily.

As it stands now, I will resort to having separate logstash instances for all my routers, but I'm looking forward to simplifying this using logstash-input-netflow. I would like to test the present version of the input, but it lacks fixes to the codec, specifically for Cisco ASR9K routers.

**Comment by ssmlt** [comment]
Hello, is there any update regarding plans to release logstash-input-netflow?
Thanks.

**Comment by jorritfolmer** [comment]
This doesn't seem relevant anymore with the addition of the non-opensource Netflow module to Logstash.

**Comment by praseodym** [comment]
What module would that be?

**Comment by jorritfolmer** [comment]
https://www.elastic.co/guide/en/logstash/current/netflow-module.html, part of x-pack apparently

**Comment by prehor** [comment]
It seems that it is part of Logstash, not X-Pack:
[https://github.com/elastic/logstash/tree/master/modules/netflow](https://github.com/elastic/logstash/tree/master/modules/netflow)
and uses standard udp input and netflow codec:
[https://github.com/elastic/logstash/blob/master/modules/netflow/configuration/logstash/netflow.conf.erb](https://github.com/elastic/logstash/blob/master/modules/netflow/configuration/logstash/netflow.conf.erb).

**Comment by acchen97** [comment]
The Logstash Netflow module is an open source feature.

**Comment by robcowart** [comment]
The logstash netflow module is nothing more than a rework of ElastiFlow 1.0.0. It is a Logstash pipeline and Kibana dashboards. The truth is that no one should be using the Logstash Netflow module. It is way behind when it comes to best leveraging the data. The pipeline uses logstash-codec-netflow and logstash-input-udp. So it has the same weakness discussed here... https://github.com/logstash-plugins/logstash-codec-netflow/issues/9

A proper Netflow input is the best solution if it fixes the above issue.

**Pull request closed without merging**